src/image/PhotoCalib.cc

```diff
@@ -293,6 +293,85 @@ MaskedImage<float> PhotoCalib::calibrateImage(MaskedImage<float> const &maskedIm
     return result;
 }
 
+afw::table::SourceCatalog PhotoCalib::calibrateCatalog(afw::table::SourceCatalog const &catalog,
+                                                       std::vector<std::string> const &instFluxFields) const {
+    auto const &inSchema = catalog.getSchema();
+    afw::table::SchemaMapper mapper(inSchema, true);  // true: share the alias map
+    mapper.addMinimalSchema(inSchema);
+
+    std::vector<table::Key<double>> instFluxKeys;
+    instFluxKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> instFluxErrKeys;
+    instFluxErrKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> fluxKeys;
+    fluxKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> fluxErrKeys;
+    fluxErrKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> magKeys;
+    magKeys.reserve(instFluxFields.size());
+    std::vector<table::Key<double>> magErrKeys;
+    magErrKeys.reserve(instFluxFields.size());
+    for (auto const &field : instFluxFields) {
+        instFluxKeys.push_back(inSchema.find<double>(field + "_instFlux").key);
```

Comment:
Once this is switched to struct field assigment (per above comment), you could actually use an even nicer syntax for this:
```
s.instFlux = inSchema[field + "_instFlux"];  // assuming s is the struct, and instFlux is a field of type Key<double>.
```
That is, with the assignment form, it can infer the fact that it's a `double` from what you're assigning it to (which is why this kind of syntax isn't usable in Python as often).

I'd also say it'd be slightly better to use `inSchema.join(field, "instFlux")` instead of `field + "_instFlux"` to avoid referring directly to the delimiter.  We don't do that across the board, but I'm trying to encourage it in new code.